cmd/bundle/generate/utils.go

```diff
@@ -50,8 +52,17 @@ func (n *downloader) markFileForDownload(ctx context.Context, filePath *string)
 		return err
 	}
 
-	filename := path.Base(*filePath)
-	targetPath := filepath.Join(n.sourceDir, filename)
+	if n.basePath == "" {
```

Comment:
Do we need to mutate state here? Would be easier to reason about if it's just a variable assignment.

Reply:
We do need state because these functions are called recursively so we need to preserve what the basePath is

Reply:
Might have missed this, but which function is recursive here?

Reply:
Sorry, I wasn't clear there, it's markDirectoryForDownload calling markFileForDownload. We could pass basePath to markFileForDownload and markNotebookForDownload but I'd prefer to incapsulate it into state of the downloader object instead